src/cookies.rs

```diff
@@ -96,7 +92,7 @@ pub trait KeyProvider {
 }
 
 #[cfg(feature = "secure")]
-impl KeyProvider for () {}
+impl<T> KeyProvider for T {}
```

Comment:
I'm unsure about this, the original idea if I remember was that if `secure` was active then the user has to implement this (by providing their own key) or use the `()` implementation to have a random key per server restart. With this change if I understand it correctly the users will by default get a random key and be unable to implement the trait for their own type without specialization?

Reply:
I see what you mean. Since we provide a default implementation for the trait, do we need to provide anything for this case? Providing an implementation for `()` requires the user to create a server data struct if they want a non-random key.

This trait cause me some confusion when I was first using nickel-cookies, because it requires my server data to implement KeyProvider, but that wasn't obvious. That was my rational for implementing it for everything. I'm not quite sure how to clarify that in the documentation, though.

Reply:
If we don't supply the implementation for `()` (the default ServerData type) then the user would have to use a custom type and implement this trait if they want to use cookies I think. And they won't be able to implement it themselves for `()` due to coherence issues.

Yeah, the documentation on this is very poor, which is I guess an artifact of it being kind of an experiment that got used :P

Reply:
Oh right. The means you can't use anything you didn't define yourself as server data. Annoying, but easy enough to work around. I will revert this.